### PR TITLE
adding new test 'test_fault_power_hit_init'

### PR DESCRIPTION
**Pull request by karolinavelkaja**
Signed-off-by: Karolina Rogowska <karolina.rogowsky@gmail.com>

### CODE REVIEW

File: test/functional/tests/fault_injection/test_fault_power_hit_init.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from api.cas import casadm, cli
+from api.cas.cache_config import CacheMode
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_utils.size import Size, Unit
+
+mount_point = "/mnt/test"
+log_path = "/var/log/opencas.log"
+wait_time_180_s = 180
+wait_time_15_s = 15
```
Comment (mmichal10) [inline test/functional/tests/fault_injection/test_fault_power_hit_init.py:17]: Unused variables

File: test/functional/tests/fault_injection/test_fault_power_hit_init.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_fault_power_hit_init(cache_mode):
+    """
+        title: Test with  power hit.
```
Comment (Ostrokrzew) [inline test/functional/tests/fault_injection/test_fault_power_hit_init.py:25]: You did what I sometimes do - You didn't modify title.

File: test/functional/tests/fault_injection/test_fault_power_hit_init.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        cache_dev.system_path = cache_device_link.get_target()
+        TestRun.executor.run_expect_success(cli.start_cmd(
+            cache_dev=str(cache_dev.system_path),
+            cache_mode=str(cache_mode.name.lower()),
+            force=True,
+            load=False))
+        TestRun.LOGGER.info(f"Successful assembly cache device with initialization")
+
+    with TestRun.step("Remove CAS device."):
+        casadm.remove_all_detached_cores()
```
Comment (Ostrokrzew) [inline test/functional/tests/fault_injection/test_fault_power_hit_init.py:58]: Is it needed? You didn't create config anywhere so cache wouldn't load at the system's start and there should be no unattached cores.

**Review by Ostrokrzew — CHANGES_REQUESTED** [review]

File: test/functional/tests/fault_injection/test_fault_power_hit_init.py (hunk 1 above)
Comment (karolinavelkaja) [reply test/functional/tests/fault_injection/test_fault_power_hit_init.py:17]: good catch, thanks!

**Review by Ostrokrzew — DISMISSED** [review]

File: test/functional/tests/fault_injection/test_fault_power_hit_init.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        core_disk.create_partitions([Size(2, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+        core_dev = core_disk.partitions[0]
+
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+        core = cache.add_core(core_dev)
+        cache_device_link = cache_dev.get_device_link("/dev/disk/by-id")
+
+    with TestRun.step("Hard reset."):
+        power_control = TestRun.plugin_manager.get_plugin('power_control')
```
Comment (katlapinka) [inline test/functional/tests/fault_injection/test_fault_power_hit_init.py:40]: you have to add @pytest.mark with required power control plugin

**Review by katlapinka — CHANGES_REQUESTED** [review]

**Review by Ostrokrzew — APPROVED** [review]

**Review by katlapinka — APPROVED** [review]

**Pull request merged**